routes/auth.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express')
+const router = express.Router()
+const {
+  sgidService,
+  scopes,
+  randomnonce,
+} = require('../lib/sgid-client.service')
+const { formatData } = require('../lib/utils')
+
+router.get('/auth/login/:env?', (req, res) => {
+  const env = req.params.env || 'prod'
+  const authUrl = sgidService[env].authorizationUrl(
+    env,
+    scopes,
+    randomnonce
+  ).url
+  res.redirect(authUrl)
+})
+
+router.get('/callback', async (req, res) => {
+  try {
+    const { code, state } = req.query
+    const { accessToken } = await sgidService[state].callback(code, randomnonce)
+
+    // api response for mobile
+    if (req.useragent.isMobile) {
```

Comment:
nit: have we considered two separate endpoints (for e.g. `/mobile-callback` vs `/callback`) for better separation of concerns? Given that the return type of the response is quite different